lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/JsonRpcProvider.kt

```diff
@@ -27,13 +27,21 @@ import kotlinx.serialization.json.*
 class JsonRpcProvider(
     val url: String,
     private val httpService: HttpService,
+    ignoreUnknownJsonKeys: Boolean,
 ) : Provider {
-    constructor(url: String) : this(url, OkHttpService())
+    private val deserializationJson = if (ignoreUnknownJsonKeys) jsonWithIgnoreUnknownKeys else Json
 
-    private val jsonWithDefaults = Json { encodeDefaults = true }
+    constructor(url: String, ignoreUnknownJsonKeys: Boolean) : this(url, OkHttpService(), ignoreUnknownJsonKeys)
+    constructor(url: String, httpService: HttpService) : this(url, httpService, false)
+    constructor(url: String) : this(url, OkHttpService(), false)
 
-    private val defaultFeeEstimateSimulationFlags: Set<SimulationFlagForEstimateFee> by lazy {
-        setOf(SimulationFlagForEstimateFee.SKIP_VALIDATE)
+    companion object {
+        private val jsonWithDefaults = Json { encodeDefaults = true }
+        private val jsonWithIgnoreUnknownKeys by lazy { Json { ignoreUnknownKeys = true } }
```

Comment:
shouldn't this be one val?

Reply:
I'm not sure which one are you referring to, both of the vals in the snippet are vals 😄

Reply:
I refer to both. Should these be two separate json objects, or one json object? if the former, why is it so?

Reply:
`jsonWithDefaults` is only used for encoding. 
`jsonWithIgnoreUnknownKeys` will be only used:
- for decoding responses
- when `ignoreUnknownJsonKeys` is true

Using one json object would cause the following problems:
- Inability to keep json instances static and available to all providers (creating new `Json` instances is a bit costly, so I think we should avoid it if possible)
- Use of `encodeDefaults` and `ignoreUnknownKeys` when these are not needed